examples/plot_metric_examples.py

```diff
@@ -0,0 +1,474 @@
+# ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+# 
+# RCA is another one of the older algorithms. It learns a full rank
+# Mahalanobis distance metric based on a weighted sum of in-class
+# covariance matrices. It applies a global linear transformation to assign
+# large weights to relevant dimensions and low weights to irrelevant
+# dimensions. Those relevant dimensions are estimated using "chunklets",
+# subsets of points that are known to belong to the same class.
+#
+# - See more in the :ref:`User Guide <rca>`
+# - See more in the documentation of the class :py:class:`RCA
+#   <metric_learn.rca.RCA>`
+
+rca = metric_learn.RCA_Supervised(num_chunks=30, chunk_size=2)
+X_rca = rca.fit_transform(X, Y)
+
+plot_tsne(X_rca, Y)
+
+######################################################################
+# Metric Learning for Kernel Regression
+# ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+#
+# An algorithm very similar to NCA but for regression is Metric
+# Learning for Kernel Regression (MLKR). It will optimize for the average
+# leave-one-out *regression* performance from a soft-nearest neighbors
+# regression.
+#
+# - See more in the :ref:`User Guide <mlkr>`
+# - See more in the documentation of the class :py:class:`MLKR
+#   <metric_learn.mlkr.MLKR>`
+#
+# To illustrate MLKR, let's use the dataset
+# `sklearn.datasets.make_regression` the same way as we did with the
+# classification  before. The dataset will contain: 100 points of 5 features
+# each, among which 3 are random noise but informative (used to generate the
+# regression target from a linear model, and two are random noise with the
+# same magnitude
+
+X_reg, Y_reg = make_regression(n_samples=100, n_informative=3, n_features=5,
+                               shuffle=True)
+
+######################################################################
+# Let's plot the dataset as is
+
+plot_tsne(X_reg, Y_reg, plt.cm.Oranges)
+
+######################################################################
+# And let's plot the dataset after transformation by MLKR:
+mlkr = metric_learn.MLKR()
+X_mlkr = mlkr.fit_transform(X_reg, Y_reg)
+plot_tsne(X_mlkr, Y_reg, plt.cm.Oranges)
+
+######################################################################
+# Points that have the same value to regress are now closer to each
+# other ! This could improve the performance of
+# `sklearn.neighbors.KNeighborsRegressor` for instance.
+
+
+######################################################################
+# Constraints
+# ^^^^^^^^^^^
+#
+# To learn the right metric, so far we have always given the labels of the
+# data to supervise the algorithms. However, in many applications,
+# it is easier to obtain information about whether two samples are
+# similar or dissimilar. For instance, when annotating a dataset of face
+# images, it is easier for an annotator to tell if two faces belong to the same
+# person or not, rather than finding the ID of the face among a huge database
+# of every person's faces.
```

Comment:
Add: "Note that for some problems (e.g., in information retrieval where the goal is to rank documents by similarity to a query document), there is no notion of individual label but one can gather information on which pairs of points are similar or dissimilar."

Reply:
That's an important point indeed, done